packages/ui-core/src/components/timezone/timezone.component.tsx

```diff
@@ -1,74 +1,136 @@
+  const { timezones, timezone, onChange } = data;
+  const tz = timezones[index];
+  const { name, utcOffset } = tz;
+  return (
+    <div style={style} role="listitem">
+      <ListItem
+        isActive={timezone && name === timezone.name}
+        onClick={() => onChange(tz)}
+      >
+        <Name>{name}</Name>
+        <Offset>{utcOffset}</Offset>
+      </ListItem>
+    </div>
+  );
 };
 
 const Timezone: FC<Props> = ({
   timezone,
+  timezones,
+  dropdownPosition = 'bottom',
+  disableSelection = false,
   timezoneLabel,
   timezonePlaceholderLabel,
   onChange,
 }) => {
   const [isOpen, setOpen] = useState(false);
-  const options = useMemo(
-    () =>
-      TIMEZONES.map(({ name }) => ({
-        label: name,
-        value: name,
-      })),
-    []
-  );
+  const listRef = useRef(null);
+
+  const options: Options = {
+    timezones,
+    timezone,
+    onChange,
+  };
+
+  useEffect(() => {
+    if (!timezone) return;
+    const index = timezones.findIndex((tz) => tz.name === timezone.name);
+
+    if (index && listRef?.current) listRef.current.scrollToItem(index, 'smart');
+  }, [timezone, isOpen]);
+
+  useEffect(() => {
+    if (isOpen && disableSelection) setOpen(false);
+  }, [disableSelection]);
 
   return (
-    <Container data-testid="timezone">
-      <Title onClick={() => setOpen(true)}>{timezoneLabel}</Title>
+    <Container isDisabled={disableSelection} data-testid="timezone">
+      <Title onClick={() => (disableSelection ? null : setOpen(true))}>
+        {timezoneLabel}
+      </Title>
       <SelectContainer>
         <DropableContainer
           variant="secondary"
           dropIndicator
-          onClick={() => !isOpen && setOpen(true)}
+          onClick={() => !isOpen && !disableSelection && setOpen(true)}
           placeholder={timezonePlaceholderLabel}
           isActive={isOpen}
           value={timezone}
           onDefocus={() => {
             setOpen(false);
           }}
         >
-          {timezone}
+          {timezone && (
+            <DropableContent>
+              <Name>{timezone.name}</Name>
+              <Offset>{timezone.utcOffset}</Offset>
+            </DropableContent>
+          )}
         </DropableContainer>
-        <Dropdown isOpen={isOpen}>
-          <DropdownListContainer scrollToActive>
-            {(activeItemRef) => (
-              <DropdownList
-                ref={activeItemRef}
-                items={options}
-                setActiveItem={({ value }) => value === timezone}
-                onClick={(_e, { value }) => {
-                  onChange(value);
-                }}
-              />
-            )}
-          </DropdownListContainer>
-        </Dropdown>
+        <AnimatePresence>
+          {isOpen && (
+            <CustomDropdown
+              position={dropdownPosition}
+              {...dropdownMotion[dropdownPosition]}
+            >
+              <ReactWindowList
+                ref={listRef}
+                height={150}
```

Comment:
Please, move it as component property `listHeight` with default value set to `150`